pkg/controller/installation/core_controller.go

```diff
@@ -325,6 +321,26 @@ func fillDefaults(instance *operator.Installation) error {
 				v6pool.BlockSize = &oneTwentyTwo
 			}
 		}
+
+		// If not specified by the user, set the flex volume plugin location based on platform.
+		if len(instance.Spec.FlexVolumePath) == 0 {
+			if instance.Spec.KubernetesProvider == operator.ProviderOpenShift {
+				// In OpenShift 4.x, the location for flexvolume plugins has changed.
+				// See: https://bugzilla.redhat.com/show_bug.cgi?id=1667606#c5
+				instance.Spec.FlexVolumePath = "/etc/kubernetes/kubelet-plugins/volume/exec/"
+			} else if instance.Spec.KubernetesProvider == operator.ProviderGKE {
+				instance.Spec.FlexVolumePath = "/home/kubernetes/flexvolume/"
+			} else if instance.Spec.KubernetesProvider == operator.ProviderAKS {
+				instance.Spec.FlexVolumePath = "/etc/kubernetes/volumeplugins/"
+			} else {
+				instance.Spec.FlexVolumePath = "/usr/libexec/kubernetes/kubelet-plugins/volume/exec/"
+			}
+		} else if instance.Spec.FlexVolumePath != "None" {
+			// Simple checks to make sure that the user provided a valid, absolute path to a directory.
+			if !path.IsAbs(instance.Spec.FlexVolumePath) {
```

Comment:
We have a separate function for validation - we should put that there rather than in `fillDefaults`